### PR TITLE
LibWasm: Implement rest of SIMD load/store instructions

### PR DESCRIPTION
**Pull request by dzfrias**

### CODE REVIEW

File: Userland/Libraries/LibWasm/AbstractMachine/BytecodeInterpreter.cpp [hunk 1]
```diff
@@ -159,6 +159,47 @@ void BytecodeInterpreter::load_and_push_mxn(Configuration& configuration, Instru
...
+    auto base = *configuration.stack().pop().get<Value>().to<u32>();
+    u64 instance_address = static_cast<u64>(bit_cast<u32>(base)) + memarg_and_lane.memory.offset;
+    Checked addition { instance_address };
+    addition += N / 8;
+    if (addition.has_overflow() || addition.value() > memory->size()) {
+        m_trap = Trap { "Memory access out of bounds" };
+        return;
+    }
+    auto slice = memory->data().bytes().slice(instance_address, N / 8);
+    auto dst = reinterpret_cast<u8*>(&vector) + memarg_and_lane.lane * N / 8;
```
Comment (alimpfard) [inline Userland/Libraries/LibWasm/AbstractMachine/BytecodeInterpreter.cpp:178]: _please_ don't use `reinterpret_cast` on pointers.
Almost every meaningful use of this is better expressed as `static_cast` or `bit_cast`. in this case `static_cast` is fine as you're not casting to a higher alignment requirement.

File: Userland/Libraries/LibWasm/AbstractMachine/BytecodeInterpreter.cpp [hunk 2]
```diff
@@ -159,6 +159,47 @@ void BytecodeInterpreter::load_and_push_mxn(Configuration& configuration, Instru
...
+    auto base = *configuration.stack().pop().get<Value>().to<u32>();
+    u64 instance_address = static_cast<u64>(bit_cast<u32>(base)) + memarg_and_lane.offset;
+    Checked addition { instance_address };
+    addition += N / 8;
+    if (addition.has_overflow() || addition.value() > memory->size()) {
+        m_trap = Trap { "Memory access out of bounds" };
+        return;
+    }
+    auto slice = memory->data().bytes().slice(instance_address, N / 8);
+    auto vector = u128();
```
Comment (alimpfard) [inline Userland/Libraries/LibWasm/AbstractMachine/BytecodeInterpreter.cpp:198]: Stylistic nitpick:
```suggestion
    u128 vector = 0;
```

File: Userland/Libraries/LibWasm/AbstractMachine/BytecodeInterpreter.cpp [hunk 3]
```diff
@@ -1716,17 +1771,31 @@ void BytecodeInterpreter::interpret(Configuration& configuration, InstructionPoi
         configuration.stack().push(Value(result));
         return;
     }
-    case Instructions::v128_any_true.value():
+    case Instructions::v128_any_true.value(): {
+        auto vector = *configuration.stack().pop().get<Value>().to<u128>();
+        configuration.stack().push(Value(static_cast<i32>(vector == u128())));
```
Comment (alimpfard) [inline Userland/Libraries/LibWasm/AbstractMachine/BytecodeInterpreter.cpp:1776]: `== 0` should be enough?

File: Userland/Libraries/LibWasm/AbstractMachine/Validator.cpp [hunk 4]
```diff
@@ -2906,7 +2906,7 @@ VALIDATE_INSTRUCTION(v128_load64_lane)
 
     TRY(validate(arg.memory.memory_index));
 
-    if (arg.memory.align > max_alignment)
+    if ((1 << arg.memory.align) > max_alignment)
         return Errors::out_of_bounds("memory op alignment"sv, 1 << arg.memory.align, 0u, max_alignment);
 
     return stack.take_and_put<ValueType::V128, ValueType::I32>(ValueType::V128);
```
Comment (alimpfard) [inline Userland/Libraries/LibWasm/AbstractMachine/Validator.cpp:2912]: Please either make this a new commit, or mention the fix in the commit description :)

**Review by alimpfard — CHANGES_REQUESTED** [review]

File: Userland/Libraries/LibWasm/AbstractMachine/BytecodeInterpreter.cpp (hunk 1 above)
Comment (alimpfard) [reply Userland/Libraries/LibWasm/AbstractMachine/BytecodeInterpreter.cpp:178]: Correction: static_cast isn't valid here, so just bit_cast instead pls.

**Comment by dzfrias** [comment]
Done, thanks for the review!

**Review by alimpfard — APPROVED** [review]

**Pull request merged**